infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -295,6 +295,7 @@ public synchronized void setRemoteLeaderNodeId(String leaderId) {
     public synchronized void resetRemoteLeaderNodeId() {
         log.debug("Reset remote leader node id");
         leaderNodeId = Optional.empty();
+        router.resetRemoteLeader(session);
```

Comment:
I wonder why we need to add resetRemoteLeader her since resetRemoteLeader is a no-op?

Reply:
It may not be a no-op operation in the client implementation of the IClientServerRouter plugin.
This method can be used to reset the stub handler if needed.

I'll elaborate more offline.